`-Wformat=` GCC 12 compiler warnings (Windows)

```
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c: In function 'ngtcp2_map_print_distance':
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c:143:21: warning: unknown conversion type character 'z' in format [-Wformat=]
  143 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c:143:58: note: format string is defined here
  143 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                                                          ^
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c:143:21: warning: unknown conversion type character 'z' in format [-Wformat=]
  143 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c:143:71: note: format string is defined here
  143 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                                                                       ^
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_map.c:143:21: warning: too many arguments for format [-Wformat-extra-args]
  143 |     fprintf(stderr, "@%u hash=%08x key=%" PRIu64 " base=%zu distance=%zu\n", i,
      |                     ^~~~~~~~~~~~~~~~~~~~~

/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_ksl.c: In function 'ksl_print':
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_ksl.c:723:24: warning: unknown conversion type character 'z' in format [-Wformat=]
  723 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                        ^
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_ksl.c:723:30: warning: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'size_t' {aka 'long l
  723 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                             ~^     ~~~~~
      |                              |     |
      |                              |     size_t {aka long long unsigned int}
      |                              unsigned int
      |                             %llu
/home/runner/work/curl-for-win/curl-for-win/ngtcp2/lib/ngtcp2_ksl.c:723:19: warning: too many arguments for format [-Wformat-extra-args]
  723 |   fprintf(stderr, "LV=%zu n=%u\n", level, blk->n);
      |                   ^~~~~~~~~~~~~~~
```
Ref: https://github.com/curl/curl-for-win/actions/runs/6711735060/job/18239769146#step:3:5624

--- a/lib/ngtcp2_ksl.c
+++ b/lib/ngtcp2_ksl.c
@@ -716,6 +716,24 @@ void ngtcp2_ksl_update_key(ngtcp2_ksl *ksl, const ngtcp2_ksl_key *old_key,
   }
 }
 
+size_t ngtcp2_ksl_len(ngtcp2_ksl *ksl) { return ksl->n; }
+
+void ngtcp2_ksl_clear(ngtcp2_ksl *ksl) {
+  if (!ksl->head) {
+    return;
+  }
+
+#ifdef NOMEMPOOL
+  ksl_free_blk(ksl, ksl->head);
+#endif /* NOMEMPOOL */
+
+  ksl->front = ksl->back = ksl->head = NULL;
+  ksl->n = 0;
+
+  ngtcp2_objalloc_clear(&ksl->blkalloc);
+}
+
+#ifndef WIN32
 static void ksl_print(ngtcp2_ksl *ksl, ngtcp2_ksl_blk *blk, size_t level) {
   size_t i;
   ngtcp2_ksl_node *node;
@@ -736,23 +754,6 @@ static void ksl_print(ngtcp2_ksl *ksl, ngtcp2_ksl_blk *blk, size_t level) {
   }
 }
 
-size_t ngtcp2_ksl_len(ngtcp2_ksl *ksl) { return ksl->n; }
-
-void ngtcp2_ksl_clear(ngtcp2_ksl *ksl) {
-  if (!ksl->head) {
-    return;
-  }
-
-#ifdef NOMEMPOOL
-  ksl_free_blk(ksl, ksl->head);
-#endif /* NOMEMPOOL */
-
-  ksl->front = ksl->back = ksl->head = NULL;
-  ksl->n = 0;
-
-  ngtcp2_objalloc_clear(&ksl->blkalloc);
-}
-
 void ngtcp2_ksl_print(ngtcp2_ksl *ksl) {
   if (!ksl->head) {
     return;
@@ -760,6 +761,7 @@ void ngtcp2_ksl_print(ngtcp2_ksl *ksl) {
 
   ksl_print(ksl, ksl->head, 0);
 }
+#endif /* !WIN32 */
 
 ngtcp2_ksl_it ngtcp2_ksl_begin(const ngtcp2_ksl *ksl) {
   ngtcp2_ksl_it it;

--- a/lib/ngtcp2_ksl.h
+++ b/lib/ngtcp2_ksl.h
@@ -263,12 +263,14 @@ void ngtcp2_ksl_clear(ngtcp2_ksl *ksl);
 #define ngtcp2_ksl_nth_node(KSL, BLK, N)                                       \
   ((ngtcp2_ksl_node *)(void *)((BLK)->nodes + (KSL)->nodelen * (N)))
 
+#ifndef WIN32
 /*
  * ngtcp2_ksl_print prints its internal state in stderr.  It assumes
  * that the key is of type int64_t.  This function should be used for
  * the debugging purpose only.
  */
 void ngtcp2_ksl_print(ngtcp2_ksl *ksl);
+#endif /* !WIN32 */
 
 /*
  * ngtcp2_ksl_it_init initializes |it|.

--- a/lib/ngtcp2_map.c
+++ b/lib/ngtcp2_map.c
@@ -126,6 +126,7 @@ static void map_bucket_set_data(ngtcp2_map_bucket *bkt, uint32_t hash,
   bkt->data = data;
 }
 
+#ifndef WIN32
 void ngtcp2_map_print_distance(ngtcp2_map *map) {
   uint32_t i;
   size_t idx;
@@ -145,6 +146,7 @@ void ngtcp2_map_print_distance(ngtcp2_map *map) {
             distance(map->tablelen, map->tablelenbits, bkt, idx));
   }
 }
+#endif /* !WIN32 */
 
 static int insert(ngtcp2_map_bucket *table, uint32_t tablelen,
                   uint32_t tablelenbits, uint32_t hash, ngtcp2_map_key_type key,

--- a/lib/ngtcp2_map.h
+++ b/lib/ngtcp2_map.h
@@ -131,6 +131,8 @@ size_t ngtcp2_map_size(ngtcp2_map *map);
 int ngtcp2_map_each(ngtcp2_map *map, int (*func)(void *data, void *ptr),
                     void *ptr);
 
+#ifndef WIN32
 void ngtcp2_map_print_distance(ngtcp2_map *map);
+#endif /* !WIN32 */
 
 #endif /* NGTCP2_MAP_H */

